### PR TITLE
Fix/434

### PR DESCRIPTION
**Pull request by FranzForstmayr**
Fixing #434 

Also merged code for step_response and impulse response

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
Is the Python2.7 build still relevant? 
unittest's assertFail was introduced in Python3 obviously.

**Comment by JAnderson419** [comment]
Re: Python 2.7. The others can correct me if I am wrong, but I believe it was decided just a week or two ago to finally drop support for 2.7, so you should be good in that department. We obviously have a bit of work to do still with CI/documentation to make that official for the next version release.

**Comment by jhillairet** [comment]
That's right, Python 2 is no more supported.

File: skrf/frequency.py [hunk 1]
```diff
@@ -318,7 +319,14 @@ def npoints(self, n):
         '''
         set the number of points in the frequency
         '''
-        self.f = linspace(self.start, self.stop, n)
+
+        if self.sweep_type == 'lin':
+            self.f = linspace(self.start, self.stop, n)
```
Comment (jhillairet) [inline skrf/frequency.py:324]: should we used 'linear' (like in matplotlib) or just 'lin' as proposed ? I'm more in favor in 'linear' because I'm used to matplotlib, but tbd

File: skrf/frequency.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 '''
 
 # from matplotlib.pyplot import gca,plot, autoscale
-from numpy import pi, linspace, logspace
+from numpy import pi, linspace, geomspace
```
Comment (jhillairet) [inline skrf/frequency.py:35]: nice conversion, I didn't know geomspace

File: skrf/frequency.py [hunk 3]
```diff
@@ -412,6 +420,15 @@ def f(self,new_f):
         '''
         self._f = npy.array(new_f)
 
+        if npy.allclose(    new_f,  
+                            linspace(new_f[0], new_f[-1], len(new_f))):
+            self.sweep_typesweep_type = 'lin'
```
Comment (jhillairet) [inline skrf/frequency.py:425]: there is a typo here I think

File: skrf/network.py [hunk 4]
```diff
@@ -3223,33 +3223,31 @@ def step_response(self, window='hamming', n=None, pad=1000):
...
+        NotImplementedError
+            If used with non equidistant sampled frequency vector
+
         See Also
         -----------
             impulse_response
             extrapolate_to_dc
         """
-        if self.nports != 1:
-            raise ValueError('Only one-ports are supported')
```
Comment (jhillairet) [inline skrf/network.py:3232]: why did you remove that part?

File: skrf/network.py [hunk 5]
```diff
@@ -3223,33 +3223,31 @@ def step_response(self, window='hamming', n=None, pad=1000):
...
-        else:
-            t = npy.flipud(npy.linspace(.5 / fstep, -.5 / fstep, n + 1, endpoint=False))
-            t = t[:-1]
-        if window != None:
-            w = self.windowed(window=window, normalize=False, center_to_dc=True)
-        else:
-            w = self
-        return t, npy.cumsum(mf.irfft(w.s, n=n).flatten())
+        t, y = self.impulse_response(window=window, n=n, pad=pad, bandpass=False)
+        return t, npy.cumsum(y)
```
Comment (jhillairet) [inline skrf/network.py:3250]: Did you check this is correct is any case (I didn't)?

File: skrf/frequency.py (hunk 1 above)
Comment (FranzForstmayr) [reply skrf/frequency.py:324]: I would prefer 'linear' too, however I didn't want to introduce a breaking change.
Personally I prefer Enums instead of strings in general.

File: skrf/network.py (hunk 4 above)
Comment (FranzForstmayr) [reply skrf/network.py:3232]: This constraint is checked in the impulse_response function.

File: skrf/network.py (hunk 5 above)
Comment (FranzForstmayr) [reply skrf/network.py:3250]: Only for an linear sweep starting at DC, otherwise an Exception is raised anyway.
Check this [gist](https://gist.github.com/FranzForstmayr/0f22a87c39aa80e7fb8b0ff3256b7a4a#file-step-py):

**Comment by FranzForstmayr** [comment]
I think I have to investigate a small issue in @f.setter tomorrow. So please wait with merging.

File: skrf/network.py (hunk 4 above)
Comment (jhillairet) [reply skrf/network.py:3232]: ha yes, OK
